tests/bridge/integration/peg_in/peg_in.rs

```diff
@@ -0,0 +1,225 @@
+use std::time::Duration;
+
+use bitcoin::{Amount, OutPoint};
+
+use bitvm::bridge::{
+    connectors::{connector::P2wshConnector, connector_0::Connector0},
+    graph::{FEE_AMOUNT, INITIAL_AMOUNT},
+    scripts::generate_pay_to_pubkey_script_address,
+    transactions::{
+        base::{BaseTransaction, Input},
+        peg_in_confirm::PegInConfirmTransaction,
+        peg_in_deposit::PegInDepositTransaction,
+        peg_in_refund::PegInRefundTransaction,
+    },
+};
+use esplora_client::Error;
+use tokio::time::sleep;
+
+use crate::bridge::{helper::generate_stub_outpoint, setup::setup_test};
+
+#[tokio::test]
+async fn test_peg_in_success() {
+    let (client, depositor_context, _, verifier_context, _, _, _, _, _, _, _, _, _) = setup_test();
+
+    let evm_address = String::from("evm address");
+
+    let input_amount_raw = INITIAL_AMOUNT + FEE_AMOUNT;
+    let deposit_input_amount = Amount::from_sat(input_amount_raw);
+
+    // peg-in deposit
+    let deposit_funding_utxo_address = generate_pay_to_pubkey_script_address(
+        depositor_context.network,
+        &depositor_context.depositor_public_key,
+    );
+    let deposit_funding_outpoint =
+        generate_stub_outpoint(&client, &deposit_funding_utxo_address, deposit_input_amount).await;
+    let deposit_input = Input {
+        outpoint: deposit_funding_outpoint,
+        amount: deposit_input_amount,
+    };
+
+    let peg_in_deposit =
+        PegInDepositTransaction::new(&depositor_context, deposit_input, evm_address.clone());
+
+    let peg_in_deposit_tx = peg_in_deposit.finalize();
+    let deposit_tx_id = peg_in_deposit_tx.compute_txid();
+
+    // mine peg-in deposit
+    let deposit_result = client.esplora.broadcast(&peg_in_deposit_tx).await;
+    assert!(deposit_result.is_ok());
+
+    // peg-in confirm
+    let confirm_funding_outpoint = OutPoint {
+        txid: deposit_tx_id,
+        vout: 0,
```

Comment:
I like to refactor this into a separate let var so we can reference it in line 55 and line 59 too

Reply:
Done